### DISCUSSION TITLE
QuickStart - minimal configuration??

### DISCUSSION THREAD

**Discussion by alport** [timmbogner/Farm-Data-Relay-System#102] [Q&A]
Hi Tim,
I have been playing with Lora and meshes for a while and have now come across your work - looks like it it could be just what I need.

I would like to get a minimal system running; Sensor sending to a MQTT gateway via ESPNow.

The problem is I don't know how to debug this - not clear if the sensor is not sending or if the gateway is not receiving.
I have compiled your examples on two D1 Mini's:

1. 1_LoRa_Sensor - with the following edit: #define GTWY_MAC      0x05
2. 5_MQTT_Gateway, with the following edits:

#define ESPNOWG_ACT    sendMQTT();  // was nothing
#define SERIAL_ACT     // sendMQTT(); 
//#include <fdrs_globals.h>

The only serial output I see is from the Gateway:
```
WiFi Connected
Connecting MQTT...
 MQTT Connected
```

Nothing is being received by my MQTT server, monitoring the topic: fdrs/#
BTW, your node-red example monitors the topic esp/fdrs which doesn't seem to be correct since you define:
#define TOPIC_DATA "fdrs/data"
Either way, I don't receive anything on esp/fdrs either!

Can you suggest any pointers?
Thanks

**Comment by timmbogner** [comment]
Hello!
The most minimal ESP-NOW setup is the one described [here](https://github.com/timmbogner/Farm-Data-Relay-System/blob/main/FDRS_Gateway/Basic_Setup.png).

- It looks like you're using the LoRa sensor to send ESP-NOW, which is your first issue.
- If you're using ESP-NOW, you cannot also use MQTT. This is due to conflicts that cause unreliability when WiFi and ESP-NOW are on different channels.  You need to split the tasks between different channels and connect the two ESP's via UART as pictured above.

```#define ESPNOWG_ACT sendMQTT();``` is not allowed.  If you want to have a LoRa to WiFi gateway that does not use ESP-NOW, _then_ you can use a single ESP (with LoRa board attached).
 ```#define LORAG_ACT sendMQTT();```  **_is_** allowed. 

Thanks for pointing the issue with my node-red example! I'm embarrassed at how long it has been incorrect.
Hope this helps. Keep me posted!
Timm

**Reply by alport** [reply]
Thanks for your quick reply.

> It looks like you're using the LoRa sensor to send ESP-NOW, which is your first issue.
My mistake - contrary to my message above I am actually using the 2_ESPNOW_Sensor which uses ESPNOW: 

```
#define USE_ESPNOW
//#define USE_LORA
```
Thanks for clarifying that ESPNow and MQTT are mutually exclusive - hence your need to bridge these two modes via the serial link.

Is your "Basic FDSR Setup" really a minimal system? Surely a simpler setup would be to only use the two ESPNow sensors (42,21) talking to the Gateway (01)? The serial output from the Gateway should then be visible on the IDE Serial monitor?
I thought I did try this - but will try it again.
Regards,

**Reply by alport** [reply]
Update:
My minimal configuration (ESPNOWsensor >> UARTGateway) works! 
Forgot the GPIO16 to reset gotcha!! I just replaced the sleepFDRS(10) with Delay(10000).
Now to add in Lora!
Thanks for your help.

**Reply by timmbogner** [reply]
That's a good point. MQTT is simpler for most uses, but the UART method is technically most minimal. There is a UART node for node-red that you can download.

The UART connection was my original go-to, but the MQTT method gives you more freedom for physically positioning the final gateway.

**Reply by timmbogner** [reply]
There are some upcoming additions and changes to FDRS's LoRa functionality that will occur in the next couple weeks. I'd reccomend switching to the 'dev' branch of the repo and start using the newer version, with LoRa packet confirmation.  There was a change made to the protocol that will make new devices incompatible with previous sensors/gateways.

**Reply by alport** [reply]
Thanks - I'll look out for the 'dev' branch.
